digest/verifiers.go

```diff
@@ -115,7 +116,7 @@ func (hv hashVerifier) Write(p []byte) (n int, err error) {
 }
 
 func (hv hashVerifier) Verified() bool {
-	return hv.digest == NewDigest(hv.digest.Algorithm(), hv.hash)
+	return strings.EqualFold(hv.digest.String(), NewDigest(hv.digest.Algorithm(), hv.hash).String())
```

Comment:
Adding case folded comparison will prevent the future use of base64,=, or other case insensitive formats. in digest target. Other code is written with the assumption that "digestA == digestB" implies their string content is identical. Let's not add complexity here. Instead, we should preprocess to canonical form where appropriate.